File: public/docs/_examples/architecture/ts/app/hero.service.ts
import { Injectable } from '@angular/core';

import { Hero } from './hero';
import { BackendService } from './backend.service';
import { Logger } from './logger.service';

@Injectable()
// #docregion class
export class HeroService {
  private heroes: Hero[] = [];

  // #docregion ctor
  constructor(
    private backend: BackendService,
    private logger: Logger) { }
  // #enddocregion ctor

  getHeroes() {
    this.backend.getAll(Hero).then( (heroes: Hero[]) => {
      this.logger.log(`Fetched ${heroes.length} heroes.`);
      this.heroes.push(...heroes); // fill cache
    });
    return this.heroes;
  }
}
// #enddocregion class

File: public/docs/_examples/architecture/dart/lib/hero_service.dart
import 'package:angular2/core.dart';

import 'backend_service.dart';
import 'hero.dart';
import 'logger_service.dart';

@Injectable()
// #docregion class
class HeroService {
  final BackendService _backendService;
  final Logger _logger;
  final List<Hero> heroes = [];

  HeroService(this._logger, this._backendService);

  List<Hero> getHeroes() {
    _backendService.getAll(Hero).then((heroes) {
      _logger.log('Fetched ${heroes.length} heroes.');
      this.heroes.addAll(heroes); // fill cache
    });
    return heroes;
  }
}
